hudi-client/hudi-java-client/src/main/java/org/apache/hudi/client/HoodieJavaWriteClient.java

```diff
@@ -226,14 +226,17 @@ protected void initMetadataTable(Option<String> instantTime, HoodieTableMetaClie
    *
    * @param inFlightInstantTimestamp - The in-flight action responsible for the metadata table initialization
    */
-  private void initializeMetadataTable(Option<String> inFlightInstantTimestamp) {
+  private void initializeMetadataTable(Option<String> inFlightInstantTimestamp, HoodieTableMetaClient metaClient) {
     if (!config.isMetadataTableEnabled()) {
       return;
     }
 
     try (HoodieTableMetadataWriter writer = JavaHoodieBackedTableMetadataWriter.create(
         context.getStorageConf(), config, context, inFlightInstantTimestamp)) {
       if (writer.isInitialized()) {
+        if (writer.isPartitionsStateChanged()) {
+          metaClient.reloadTableConfig();
+        }
```

Comment:
Should we also update Flink write client with the same table config reload logic?